tests/e2e/test_rosacli_iam_service_account.go

```diff
@@ -0,0 +1,364 @@
+/*
+Copyright (c) 2025 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"fmt"
+	"math/rand"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"github.com/openshift/rosa/tests/ci/labels"
+	"github.com/openshift/rosa/tests/utils/exec/rosacli"
+	. "github.com/openshift/rosa/tests/utils/helper"
+)
+
+var _ = Describe("IAM Service Account", labels.Feature.IAMServiceAccount, func() {
+	defer GinkgoRecover()
+
+	var (
+		rosaClient *rosacli.Client
+		// Track resources for cleanup
+		serviceAccountRolesToClean []string
+		testClusterID              string
+		// Generate unique namespace for tests
+		testNamespace string
+	)
+
+	BeforeEach(func() {
+		By("Initialize the client")
+		rosaClient = rosacli.NewClient()
+
+		By("Get a test cluster")
+		clusterOutput, err := rosaClient.Cluster.List()
+		Expect(err).ToNot(HaveOccurred())
+		clusters, err := rosaClient.Cluster.ReflectClusterList(clusterOutput)
+		Expect(err).ToNot(HaveOccurred())
+
+		// Find an STS cluster for testing
+		var testCluster *rosacli.ClusterDescription
+		found := false
+		for _, cluster := range clusters.Clusters {
+			// Get detailed cluster description to check if it's an STS cluster
+			clusterDesc, err := rosaClient.Cluster.DescribeClusterAndReflect(cluster.ID)
+			Expect(err).ToNot(HaveOccurred())
+
+			if clusterDesc.STSRoleArn != "" {
+				testCluster = clusterDesc
+				testClusterID = cluster.ID
+				found = true
+				break
+			}
+		}
+
+		if !found {
+			Skip("No STS cluster available for testing")
```

Comment:
I don't think we should skip if no STS clusters are found- we want all e2e tests to run when their flag is invoked